python/GafferSceneUI/SetUI.py

```diff
@@ -223,13 +229,15 @@ def __popupMenu( menuDefinition, plugValueWidget ) :
 	if plug is None :
 		return
 
+	targetPlug = __targetPlug( plug ) or plug
```

Comment:
I think we can return here if the target plug was `None`, and remove the return at L242.

Reply:
Removing the return at L242 would miss the condition `hasTextWidget` which is part of `acceptsSetExpression`. The `Sets` menu setup uses a couple of fall-through cases that assume a valid `currentNames` value, which I could rework. It seems like keeping the return at L242 is a little more readable though?

Reply:
Ah, right you are. I think we still need a return here a well though, so we don't go making metadata queries for a `None` targetPlug?

Reply:
Yes, that early return will avoid querying metadata for a `None` plug, fixed in ca33c0454ee8613bdebbc4601f55c98263829ca0